### ISSUE TITLE
"Ping-pong keep alive missing on time lead" to "throttle queue is over maxCapacity" in bybit ccxt pro on watchOHLC

### ISSUE THREAD

**Issue by gaugau3000** [ccxt/ccxt#23929]
### Operating System

Linux

### Programming Languages

JavaScript

### CCXT Version

4.4.10

### Description

Workflow : 

1/ Connection to 416 bybit linear watch candle on ccxt pro on 1m timeframe.
2/ Loose ping-pong on the 416 ws conn.

`Error with : Connection to wss://stream.bybit.com/v5/public/linear timed out due to a ping-pong keepalive missing on time for THETA/USDT:USDT`

3/ the retry lead to:

`Error with : throttle queue is over maxCapacity (1000), see https://github.com/ccxt/ccxt/issues/11645#issuecomment-1195695526 for GOMINING/USDT:USDT`

4/ the program lead to memory leak.

Here I am not sure why the queue is over capacity because the retry should not really lead to increase it (keep at 416) ?

Many TKS.

### Code

```js
async wsCandlesSubscribe(candlesManagersForPair: CandlesManager[], relatedPair: string): Promise<void> {
        while (true) {
            try {
                const snapshot1mCandles = await this.ccxtExchange.watchOHLCV(relatedPair, "1m") as OHLCV[];
                for (const candlesManagerForPairTf of candlesManagersForPair) {
                    this.processCandle1mSnapshot(snapshot1mCandles, candlesManagerForPairTf);
                }
            } catch (error) {
                this.handleError(error, relatedPair);
            }
        }
    }
```

**Comment by carlosmiei** [comment]
Hello @gaugau3000,

Thanks for reporting. Instead of calling `watchOHLCV` 416 times, can you try to use `watchOHLCVForSymbols` instead?

It should be used like `ex.watchOHLCVForSymbols([[symbol, timeframe], [symbol2, params2], params)`

**Comment by carlosmiei** [comment]
Nonetheless I will try to check the issue you're facing.

**Comment by gaugau3000** [comment]
> Hello @gaugau3000,
> 
> Thanks for reporting. Instead of calling `watchOHLCV` 416 times, can you try to use `watchOHLCVForSymbols` instead?
> 
> It should be used like `ex.watchOHLCVForSymbols([[symbol, timeframe], [symbol2, params2], params)`

Yes, will give a try, many TKS (I open one connection for each symbol vs on connection with n symbols with is poor choice design on my side maybe).

**Comment by carlosmiei** [comment]
@gaugau3000 The connection is always reused so you have only one connection regardless of the approach, however sending hundreds of messages might be a problem.

**Comment by kroitor** [comment]
Hello @gaugau3000, closing this during a cleanup — the answer above remains the resolution: with 416 independent watch loops, a single keepalive miss makes every loop resubscribe at once, and those hundreds of throttled sends are exactly what overflows the queue guard and churns memory during the retry storm. `watch_ohlcv_for_symbols` subscribes the same set with a handful of batched messages over the same shared connection, which removes the storm on both initial connect and every reconnect — it's the intended tool for watchlists of this size.

For very large lists two knobs help further: splitting the symbols across a few batched subscriptions rather than one giant one, and staggering your outer retry loops with a small jitter so a reconnect never fires everything in the same tick. If the batched method itself misbehaves at scale on a current version, a fresh issue with a verbose log would be very welcome. Thanks!
